pkg/kube/podtemplatespec/podspec_template_test.go

```diff
@@ -342,6 +342,57 @@ func TestMergeVolumeMounts(t *testing.T) {
 	assert.Equal(t, []corev1.VolumeMount{vol2, vol0}, mergedVolumeMounts)
 }
 
+func TestMergeVolumesSecret(t *testing.T) {
+	permission := int32(416)
+	vol0 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "Secret-name"}}}}
+	vol1 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{DefaultMode: &permission}}}}
+	mergedVolumes := mergeVolumes(vol0, vol1)
+	assert.Len(t, mergedVolumes, 1)
+	volume := mergedVolumes[0]
+	assert.Equal(t, "volume", volume.Name)
+	assert.Equal(t, corev1.SecretVolumeSource{SecretName: "Secret-name", DefaultMode: &permission}, *volume.Secret)
+}
+
+func TestMergeNonNilValueNotFilledByOperator(t *testing.T) {
+	// Tests that providing a custom volume with a volume source
+	// That the operator does not manage overwrites the original
+	vol0 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "Secret-name"}}}}
+	vol1 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{GCEPersistentDisk: &corev1.GCEPersistentDiskVolumeSource{}}}}
+	mergedVolumes := mergeVolumes(vol0, vol1)
+	assert.Len(t, mergedVolumes, 1)
+	volume := mergedVolumes[0]
+	assert.Equal(t, "volume", volume.Name)
+	assert.Equal(t, corev1.GCEPersistentDiskVolumeSource{}, *volume.GCEPersistentDisk)
+	assert.Nil(t, volume.Secret)
+}
+
+func TestMergeNonNilValueFilledByOperatorButDifferent(t *testing.T) {
+	// Tests that providing a custom volume with a volume source
+	// That the operator does manage, but different from the one
+	// That already exists, overwrites the original
+	vol0 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "Secret-name"}}}}
+	vol1 := []corev1.Volume{{Name: "volume", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}}}
+	mergedVolumes := mergeVolumes(vol0, vol1)
+	assert.Len(t, mergedVolumes, 1)
+	volume := mergedVolumes[0]
+	assert.Equal(t, "volume", volume.Name)
+	assert.Equal(t, corev1.EmptyDirVolumeSource{}, *volume.EmptyDir)
+	assert.Nil(t, volume.Secret)
+}
+
+func TestMergeVolumeAddVolume(t *testing.T) {
+	vol0 := []corev1.Volume{{Name: "volume0", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{}}}}
+	vol1 := []corev1.Volume{{Name: "volume1", VolumeSource: corev1.VolumeSource{EmptyDir: &corev1.EmptyDirVolumeSource{}}}}
+	mergedVolumes := mergeVolumes(vol0, vol1)
+	assert.Len(t, mergedVolumes, 2)
+	volume0 := mergedVolumes[0]
+	assert.Equal(t, "volume0", volume0.Name)
+	assert.Equal(t, corev1.SecretVolumeSource{}, *volume0.Secret)
+	volume1 := mergedVolumes[1]
+	assert.Equal(t, "volume1", volume1.Name)
+	assert.Equal(t, corev1.EmptyDirVolumeSource{}, *volume1.EmptyDir)
+}
```

Comment:
Probably unnecessary, but I wanted to also have a "simple" tests that just performs the very basic